File: backend/models/trip.py
from sqlalchemy import Column, Integer, String, Float, Text, ForeignKey
from sqlalchemy.orm import relationship
from database import Base


class Trip(Base):
    __tablename__ = "trips"

    id                = Column(Integer, primary_key=True)
    user_id           = Column(Integer, ForeignKey("users.id"), nullable=False)
    destination       = Column(String,  nullable=False)
    days              = Column(Integer, nullable=False)
    budget            = Column(Float,   nullable=False)
    category          = Column(String,  nullable=False)
    daily_budget      = Column(Float,   nullable=False)
    travel_style      = Column(String,  nullable=True, default="balanced")
    ai_recommendation = Column(Text,    nullable=True)

    owner = relationship("User", back_populates="trips")

File: frontend/types/trip.ts
export interface TripData {
  id?: number;
  destination?: string;
  days?: number;
  budget?: number;
  travel_style?: string;
  itinerary?: string;
  created_at?: string;
  [key: string]: any;
}

export interface FormValues {
  destination: string;
  days: number;
  budget: number;
  travel_style: string;
}
